### ISSUE TITLE
JSON Schema & Pydantic's default factory.

### ISSUE THREAD

**Issue by koxudaxi** [koxudaxi/datamodel-code-generator#1026]
### Discussed in https://github.com/koxudaxi/datamodel-code-generator/discussions/882

<div type='discussions-op-text'>

<sup>Originally posted by **zcherry17** October 24, 2022</sup>
Unless this is already a feature, I cannot find anything in the code base that supports this.
I am trying to make a schema that generates the following:

```python
class Foo(BaseModel):
    my_list: list[int] = Field(default_factory=list)
```

Firstly, is there a reason why default factory is not supported? If there is not any particular reason, then a default factory should be generated for arrays in some meaningful way. Perhaps if the following schema was defined, then you would get the appropriate generation.

```json
{
    "$id": "/foo",
    "$schema": "https://json-schema.org/draft/2020-12/schema",
    "title": "Foo",
    "type": "object",
    "properties": {
        "my_list": {
            "type": "array",
            "items": {"type": "integer"}
            "default": { "default_factory": "list" }
            < -- or -- >
            "default_factory": "list" < -- as I have seen that extra keys are supported in the code generator -->
        }
    }
}
```
</div>

**Comment by tommyjcarpenter** [comment]
is the first variant here `"default": { "default_factory": "list" }` supported? i just found out prance complains about the latter one, because default_factory isnt a part of the json schema i guess

**Comment by koxudaxi** [comment]
@tommyjcarpenter 
No.codegen don't support the structure of default_factry.
In OpenAPI, we can use `x-` prefix for the original key.
But, JsonSchema may not provide a special way.

**Comment by tommyjcarpenter** [comment]
oh, let me try the `x-`! that would be perfect.

**Comment by tommyjcarpenter** [comment]
<img width="981" alt="Screenshot 2023-04-26 at 1 37 01 PM" src="https://user-images.githubusercontent.com/9419125/234657543-9f0c76f0-995f-4649-9a8e-6b67c4f3e730.png">

adding x- makes openapi pass, but appears to not work with this repo lol

could you support `x-default-factory`?

**Comment by koxudaxi** [comment]
We have the option :wink: 
```bash
  --field-extra-keys-without-x-prefix
                        Add extra keys with `x-` prefix to field parameters.
                        The extra keys are stripped of the `x-` prefix.
```

**Comment by tommyjcarpenter** [comment]
@koxudaxi ok i think i did find an issue

using that option with --strict-nullable leads back to this issue: https://github.com/koxudaxi/datamodel-code-generator/issues/1216

<img width="464" alt="Screenshot 2023-04-26 at 5 27 40 PM" src="https://user-images.githubusercontent.com/9419125/234706922-e6903425-5147-48a3-8cb1-7a6ac021a45c.png">

->

<img width="628" alt="Screenshot 2023-04-26 at 5 28 25 PM" src="https://user-images.githubusercontent.com/9419125/234707059-05f03761-9bad-4aee-97b2-c20e0707d78c.png">

**Comment by koxudaxi** [comment]
@tommyjcarpenter 
I just released `0.18.1`
Could you please try this version?
It has the fixes.
https://github.com/koxudaxi/datamodel-code-generator/pull/1254
